### PR TITLE
Feat/generate links

### PR DESCRIPTION
**Pull request by ciaranschutte**
- adds script to generate docSitePaths file for platform ui

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "add": "node manage-versions.js -a --npm --silent",
     "clean": "node clean-data.js",
     "test": "mocha --inline-diff",
-    "watch-tests": "mocha ./**/*.test.js --inline-diff -w"
+    "watch-tests": "mocha ./**/*.test.js --inline-diff -w",
+    "generate-platform-links": "node links/generate.js"
```
Comment (joneubank) [inline scripts/package.json:9]: just call this `links`... make everyone happy forever.

Comment (ciaranschutte) [reply scripts/package.json:9]: it's that simple? I wish I had known this earlier!

File: scripts/readme.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Generate data
+
+# Generate links
```
Comment (joneubank) [inline scripts/readme.md:3]: Thanks for documenting. Does the script just console out the new values or do you need to document where it stores the values?

Do we need to document in a deployment document for the platform how to do this generation?

File: scripts/links/generate.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const links = await Promise.all(
+    startingPaths.map(async (pathEntry) => {
+      const resPath = resolve(path, pathEntry.name);
+      return pathEntry.isDirectory() ? traverse(resPath) : generateLink(resPath);
+    }),
+  );
+  return Array.prototype
+    .concat(...links)
+    .filter(Boolean)
+    .join('');
```
Comment (hlminh2000) [inline scripts/links/generate.js:24]: wanna template in a comment at the top with a comment mentioning this file? :)

Comment (ciaranschutte) [reply scripts/links/generate.js:39]: @hlminh2000 i made it medium scary

**Pull request merged**